week4.Rmd

```diff
@@ -0,0 +1,236 @@
+
+```{r, include = FALSE, message = FALSE}
+# required behind the scenes to find the correct python installation
+library(reticulate)
+
+# include the following command to activate a specific conda environment:
+#use_condaenv(condaenv = "my_env", required = TRUE)
+
+# note that this may get stuck knitting automatically in RStudio but will work 
+# if you knit manually using the following command from the console:
+#rmarkdown::render("example_python_in_rstudio.Rmd")
+```
+
+```{python, include = FALSE}
+# load IPython library for comprehensive system info
+import IPython
+import re
+print(IPython.sys_info())
+```
+
+# Learning Goals
+- Import and parse data in python
+- Use functions to manipulate DNA sequences with python
+- Understand how plasmid assembly can be done virtually
+- Apply the basics of DNA ligation/restriction enzymes
+
+# Prerequisites
+- Basics of how plasmids are assembled with restriction enzyme digests
+- Basics of bacteria and DNA
+- Basic python syntax
+
+# Background
+Every day, researchers assemble and edit DNA for an uncountable variety of purposes. In the field of Synthetic Biology, researchers often manipulate the DNA of a microorganism in order to change its phenotype, enabling the organism to conduct behavior it otherwise would not have been able to. The canonical example of this is the creation of a circular DNA assembly, known as a *plasmid*, that contains the human gene for insulin. This plasmid is then inserted into a culture of bacteria, often **E. coli**, and produced on an industrial scale.
+
+But how does one create a plasmid? A plasmid must be able to be "read" by its host bacteria and transcribed into RNA, then the target protein. The challenge here isn't in **synthesizing** the DNA (this can be done in an industrial lab at a relatively low cost), but rather in designing a plasmid logically.
+
+A functional plasmid has a few important components:
+1. A vector backbone that allows insertion/proliferation of the plasmid while also providing antibiotic resistance (for genetic screening)
+2. The gene of interest (GOI)
+
+
+```{python, include = FALSE}
+goi = "ATGAACAATGCACTATGGTCAAAAATGCTGGTGGAAGAGGATCCCGGGTTTTTTAGACATAGATACACTCATCAGCAACAATGCACTATGGTCAAAAATGCTGGTGGAAGAGGATCCCGGGTTTTTTGAGAAACTG"
+backbone = "aggctcagtcgaaagactgggcctttcgttttatctgttgtttgtcggtgaacgctctctactagagtcacactggctcaccttcgggtgggcctttctgcgtttataccttaaatattggtttaaaagcaggttaaaagacaggttagcggtggccgaaaaacgggcggaaacccttgcaaa"
+```
+
+In a hidden code chunk I've loaded in a GOI and vector backbone as `goi`,`backbone`, respectively. You can use a `return` call of these variables to print their values.
+
+NOTE: For ease of viewing, these gene sequences have been significantly abbreviated.
+
+If you've printed `backbone`, you may notice that some of the gene sequence is written in upper-text and some in lower-case. Let's write a function to convert all our genes to upper-case.
+
+```{python}
```

Comment:
Would be useful to see comments for each line of code so that students can follow along (plus I'm not so familiar with python lingo, so it would help me too!)

Reply:
Added additional comments to walk through the python used here.